src/plugins/controls/connectivity/connectivity.js

```diff
@@ -223,6 +223,13 @@ export default pluginFactory({
                 });
                 return false;
             }
+
+            if (proxy.isOffline() && item.flags && item.flags.hasPci) {
+                self.displayWaitingDialog().then(function () {
+                    testRunner.loadItem(itemRef);
+                });
+                return false;
+            }
```

Comment:
**nitpick:** This code is identical to the previous block. It could be refactored.

**suggestion:** As the only difference is the flag to check, consider factor this. For instance, this could be done like this, to replace the whole listener:
```javascript
         testRunner.before('loaditem.connectivity', function (e, itemRef, item) {
            const testContext = testRunner.getTestContext();
            const { flags } = item;

            if (flags) {
                if (flags.hasFeedbacks) {
                    testContext.hasFeedbacks = true;
                }

                if (proxy.isOffline()) {
                    const waitingFlags = ['containsNonPreloadedAssets', 'hasPci'];
                    const needToWait = waitingFlags.some(flag => flags[flag]);
                    if (needToWait) {
                        self.displayWaitingDialog().then(function () {
                            testRunner.loadItem(itemRef);
                        });
                        return false;
                    }
                }
            }
        });
```